xdsl/traits.py

```diff
@@ -94,6 +94,66 @@ def verify(self, op: Operation) -> None:
                 )
 
 
+class SingleBlockImplicitTerminator(OpTrait):
+    """
+    Checks and adds if missing the specified terminator to an operation which has
+    single-block region.
+
+    This should be fully compatible with MLIR's Trait:
+    https://mlir.llvm.org/docs/Traits/#single-block-with-implicit-terminator
+    """
+
+    parameters: type[Operation]
```

Comment:
Why parameterS? Why not just parameter? This is always a single operation, right? I guess we could think about building some form of constraint from multiple ops (say "any_of"), but even in that case I would argue this is only one parameter.

Reply:
Because `OpTrait` has a `parameters` member and I'm repurposing that via inheritance.